include/seqan3/alignment/pairwise/detail/type_traits.hpp

```diff
@@ -83,6 +85,12 @@ template <typename configuration_t>
 struct alignment_configuration_traits
 {
 private:
+    //!\brief Transforms integral types to unsigned types and chooses the minimal viable uint for floating point types.
+    template <arithmetic score_t>
+    using select_scalar_index_t = lazy_conditional_t<std::integral<score_t>,
+                                                     lazy<std::make_unsigned_t, score_t>,
+                                                     min_viable_uint_t<1ull << (sizeof_bits<score_t> - 1)>>;
```

Comment:
This does not seem like a float. Okay you somehow transform a float into a uint of the same bit size.

Reply:
Why? This seems extremely unrelated?

Reply:
Because I want the index type to be an unsigned type. And if the score type is a floating point I need to know the corresponding uint type that has the same byte size as float or double if I put it into a simd vector.

Reply:
Now that I understand the purpose, I think this should be a property of the `simd_traits`.

We already have

```c++
using int32x_t = seqan3::simd_type_t<int>;
typename seqan3::simd_traits<int32x_t>::mask_type; // same as int32x_t{} == int32x_t{};
typename seqan3::simd_traits<int32x_t>::swizzle_type; // same as seqan3::simd_type_t<uint8_t>;
```

We should add a

```c++
typename simd_traits<int32x_t>::rebind<new_scalar_t> //  same length, same max_length, but with new scalar_t; gives nice compiler error if new scalar t has not same bit size as 
```

and should also add a `index` structure that is compatible to the simd vector.

```c++
typename seqan3::simd_traits<int32x_t>::index_type; // simd_traits<int_simd_t>::rebind<unsigned_of_same_bit_size>
```

We keep your changes here, and add this as an "easy" task for a next sprint

Reply:
No the index_type has nothing todo with the simd traits type. I would not add this to the simd traits class. I like the idea of the rebind though. For the latter I suggest if needed somewhere else we have an explicit transformation trait for this. But to cite you here: `it does not follow our policy of adding stuff only if it is needed.` That however does not resolve the problem of identifying the right index type based on the input type. So suggestion would be to have an explicit tranformation trait:

```cpp
template <typename t>
struct index_type_for;

template <std::integral t>
struct index_type_for<t> 
{ 
    using type = std::make_unsigned_t<t>; 
} ;

template <std::floating_point t>
struct index_type_for<t> 
{ 
    using type = min_viable_uint_t<1ull << (sizeof_bits<score_t> - 1)>; 
} ;

template <arithmetic t>
using index_type_for_t = typename index_type_for<t>::type;
```

Of course we could also put the two overloads together but maybe it can be better described of what happens here. If we are putting it out anyway I would prefer the more verbose version over the shorter version.

Reply:
> No the index_type has nothing todo with the simd traits type. I would not add this to the simd traits class. I like the idea of the rebind though. For the latter I suggest if needed somewhere else we have an explicit transformation trait for this.

I disagree here, because we already have `mask_type` and `swizzle_type`, which are transformation traits specific to a simd vector. It would make sense to add a way to give you a `index_type`. But let's discuss this later.

Reply:
But, swizzle type and mask type are entities that are specific to the intrinsics. For some instructions like AVX512 one could wrongly think that the index type is something you provide for the shuffle operation which is then a special non-simd register type. But we want to express inside of one algorithm, that we want the corresponding unsigned type of the score type (and in case of the float/double the corresponding unsigned type). This can be achieved already with the rebind but is not a property of the simd vector itself.